[Video Capture] Audio alerts for recording timer

manikd31 opened the pull request:
## AV-3983: Audible Alerts for Video-Capture script
- Ticket: [AV-3983](https://20bn.atlassian.net/browse/AV-3983)

### Developer Notes:
This PR aims at adding audible alerts to the `video_capture.py` script. In addition to the visual timer for countdown, audio-alerts will be played in both, pre-recording and recording phases with the following conditions:
- **Pre-recording**: Count down (alert) the last three seconds of the timer, and another alert for pre-record phase end, i.e. start recording.
- **Recording**: Just a single alert when the recording phase has ended.

**Usage**: Audio-alerts are by default enabled. To disable these, add a flag for `--no_audio` while calling `video_capture.py`

**Edit (from Annotation)**: Changed text thickness on video-prompt to be more visible to the user

sunny-panchal commented on scripts/tools/video_capture.py:
```diff
@@ -25,9 +25,43 @@
+        self.shutdown_fn = shutdown_fn
+
+    def run(self):
+        print("THIS IS A DEBUG PRINT TO SEE IF RUN IS BEING CALLED SUCCESSFULLY")
```
leftover

manikd31 replied:
Removed this class, was not being used.

sunny-panchal commented on scripts/tools/video_capture.py:
```diff
@@ -25,9 +25,43 @@
 
 FONT = cv2.FONT_HERSHEY_PLAIN
 _shutdown = False
+countdown_sound = 'counter_sound.wav'
```
constant should be all caps

manikd31 replied:
Done.

sunny-panchal commented on scripts/tools/video_capture.py:
```diff
@@ -46,24 +80,32 @@ def _capture_video(video_duration=0., record=False):
 
             if record:
-                message = f"recording video {str(index + 1)}"
+                message = f"Recording Video {str(index + 1)}"
```
```suggestion
                message = f"Recording video {str(index + 1)}"
```

manikd31 replied:
Done.

sunny-panchal reviewed (commented):
Great! It works but I think there are still a few things to address:
- add required packages to `requirements.txt`
- refactor media files to either:
    - `docs/audio/...`
    - `scripts/tools/video_capture/...`
    - The tones play over "2, 1, 0, end." Should it be "3, 2, 1, end"? displayed countdown values should be incremented by 1
    - I think the tone for countdown and done sound better swapped

sunny-panchal commented on scripts/tools/video_capture.py:
```diff
@@ -25,9 +25,43 @@
+done_sound = 'long_done_sound.wav'
+
+
+class ShutDownWatcher(Thread):
```
Also, this class is unused

manikd31 replied:
Removed.

manikd31 commented:
> Great! It works but I think there are still a few things to address:
> 
>     * add required packages to `requirements.txt`
> 
>     * refactor media files to either:
>       
>       * `docs/audio/...`
>       * `scripts/tools/video_capture/...`
>       * The tones play over "2, 1, 0, end." Should it be "3, 2, 1, end"? displayed countdown values should be incremented by 1
>       * I think the tone for countdown and done sound better swapped

Addressed all comments and suggestions, apart from swapping `done` and `countdown` sounds. Do we need more opinions on that? I personally feel having a high-pitch sounds for `Done` makes sense, but we can swap that anytime.

manikd31 commented:
> > Great! It works but I think there are still a few things to address:
> > ```
> > * add required packages to `requirements.txt`
> > 
> > * refactor media files to either:
> >   
> >   * `docs/audio/...`
> >   * `scripts/tools/video_capture/...`
> >   * The tones play over "2, 1, 0, end." Should it be "3, 2, 1, end"? displayed countdown values should be incremented by 1
> >   * I think the tone for countdown and done sound better swapped
> > ```
> 
> Addressed all comments and suggestions, apart from swapping `done` and `countdown` sounds. Do we need more opinions on that? I personally feel having a high-pitch sounds for `Done` makes sense, but we can swap that anytime.

Updated sounds for `done` and `countdown`. Also, added another sound for `exit`.

sunny-panchal reviewed (approved):
Had to increase the margin otherwise the countdown wouldn't play on a few trials otherwise looks good :+1:

florianletsch commented:
@sunny-panchal @manikd31 Where are the audio files from? What's the license? We have to include the appropriate license disclaimer in the README because binary files are usually not licensed under MIT.

sunny-panchal commented:
> @sunny-panchal @manikd31 Where are the audio files from? What's the license? We have to include the appropriate license disclaimer in the README because binary files are usually not licensed under MIT.

Good point... @manikd31 is rejoining us tomorrow and I'll confirm with him.
